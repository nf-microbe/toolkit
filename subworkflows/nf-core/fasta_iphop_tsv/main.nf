// Import modules
include { IPHOP_DOWNLOAD    } from '../../../modules/nf-core/iphop/download'
include { IPHOP_PREDICT     } from '../../../modules/nf-core/iphop/predict'

workflow FASTA_IPHOP_TSV {
    take:
    fasta_gz    // [ [ meta ], fasta.gz ]
    iphop_db    // [ iphop_db ]

    main:
    ch_versions             = Channel.empty()

    // if iphop_db exists, skip IPHOP_DOWNLOAD
    if (iphop_db){
        ch_iphop_db = iphop_db
    } else {
        //
        // MODULE: download iPHOP database
        //
        IPHOP_DOWNLOAD(
        )
        ch_iphop_db = IPHOP_DOWNLOAD.out.iphop_db
        ch_versions = ch_versions.mix(IPHOP_DOWNLOAD.out.versions)
    }

    //
    // MODULE: Run geNomad end-to-end
    //
    IPHOP_PREDICT(
        fasta_gz,
        ch_iphop_db
    )
    ch_versions = ch_versions.mix(IPHOP_PREDICT.out.versions)

    emit:
    ch_iphop_genus_tsv  = IPHOP_PREDICT.out.iphop_genus     // [ [ meta ], Host_prediction_to_genus_m*.csv ]
    ch_iphop_genome_tsv = IPHOP_PREDICT.out.iphop_genome    // [ [ meta ], Host_prediction_to_genome_m*.csv ]
    versions            = ch_versions                       // channel: [ versions.yml ]
}
