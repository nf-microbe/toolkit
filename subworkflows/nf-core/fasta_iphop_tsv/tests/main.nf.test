nextflow_workflow {

    name "Test Subworkflow FASTA_IPHOP_TSV"
    script "../main.nf"
    workflow "FASTA_IPHOP_TSV"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/fasta_iphop_tsv"
    tag "iphop"
    tag "iphop/download"
    tag "iphop/predict"

    test("sarscov2 - genome - fasta.gz") {
        when {
            params {
                iphop_download_args = "-dbv iPHoP_db_rw_for-test"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test1' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta.gz', checkIfExists: true)
                    ],
                    [
                        [ id:'test2' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true)
                    ]
                )
                input[1] = null
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
