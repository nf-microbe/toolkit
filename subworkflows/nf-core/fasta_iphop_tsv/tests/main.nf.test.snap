{
    "sarscov2 - genome - fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_Host_prediction_to_genus.csv:md5,6f80d41b3a5ef039141effe4b644bb65"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_Host_prediction_to_genus.csv:md5,80ccdf83ce025aafc57fdef5a843f697"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_Host_prediction_to_genome.csv:md5,ff5e6a01204756c5f7eca4fc050f41c0"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_Host_prediction_to_genome.csv:md5,946202be48d61b4961fe2f541e0ec256"
                    ]
                ],
                "2": [
                    "versions.yml:md5,39a45cab5b23795bf4ee316052acf29e",
                    "versions.yml:md5,5dbdeebc6b88051af651b5b7cdeb2254",
                    "versions.yml:md5,5dbdeebc6b88051af651b5b7cdeb2254"
                ],
                "ch_iphop_genome_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_Host_prediction_to_genome.csv:md5,ff5e6a01204756c5f7eca4fc050f41c0"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_Host_prediction_to_genome.csv:md5,946202be48d61b4961fe2f541e0ec256"
                    ]
                ],
                "ch_iphop_genus_tsv": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_Host_prediction_to_genus.csv:md5,6f80d41b3a5ef039141effe4b644bb65"
                    ],
                    [
                        {
                            "id": "test2"
                        },
                        "test2_Host_prediction_to_genus.csv:md5,80ccdf83ce025aafc57fdef5a843f697"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,39a45cab5b23795bf4ee316052acf29e",
                    "versions.yml:md5,5dbdeebc6b88051af651b5b7cdeb2254",
                    "versions.yml:md5,5dbdeebc6b88051af651b5b7cdeb2254"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-15T16:49:34.267980443"
    }
}